src/models/agents/basic-animal.js

```diff
@@ -182,7 +183,11 @@ export default class BasicAnimal extends Agent {
     const newLoc = {x: loc.x + dx, y: loc.y + dy};
     if (this.environment.crossesBarrier(loc, newLoc)) {
       // stay where you are and turn around, for now
-      return this.set('direction', dir + Math.PI);
+      if (Math.cos(dir) > 0) {
```

Comment:
Trying to keep `direction` in the 0 - 2PI range for future comparison.

Reply:
:+1: though a general solution could also be on the table, since nothing is preventing speed being set out of these bounds in an animal's overwritten `wander` method or anywhere else. We could possibly put validation logic in the `set` method for values that ought to be bounded.

Reply:
For the general case I believe this shouldn't affect anything. And to clarify, I'm not really trying to stay within 0 to 2PI, but more like -PI/4 to 2PI (essentially limiting really big or small changes that affect later comparison).  The most simple thing really would be to adjust my comparison so it uses degrees instead or radians.  Then all values would be acceptable for `direction` and we could revert to the previous code.